src/main/java/bio/terra/service/common/JournalRecordCreateEntryStep.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.service.common;
+
+import static bio.terra.service.common.CommonFlightUtils.getFlightInformationOfInterest;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.auth.iam.IamResourceType;
+import bio.terra.service.journal.JournalService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public abstract class JournalRecordCreateEntryStep implements Step {
+  private final JournalService journalService;
+  private final AuthenticatedUserRequest userReq;
+  private final IamResourceType resourceType;
+  private final String note;
+  private final boolean clearHistory;
+
+  public JournalRecordCreateEntryStep(
+      JournalService journalService,
+      AuthenticatedUserRequest userRequest,
+      IamResourceType resourceType,
+      String note,
+      boolean clearHistory) {
+    this.journalService = journalService;
+    this.userReq = userRequest;
+    this.resourceType = resourceType;
+    this.note = note;
+    this.clearHistory = clearHistory;
+  }
+
+  public abstract UUID getResourceId(FlightContext context) throws InterruptedException;
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    journalService.recordCreate(
```

Comment:
Not sure it's worth changing for this PR, but the only reason there needs to be three different steps is because `JournalService` doesn't have an API that lets a caller pass in `JournalService.EntryType`. If it did, you could make the entry type a constructor argument to the step. And maybe have three different factory methods so the caller doesn't need to know the type values.

Reply:
Thanks for the suggestion! It would be nice to get this down to a single step but I'm inclined to leave it as-is for now!